test/azure/scanner.spec.ts

```diff
@@ -1,24 +1,70 @@
 import {assert} from 'chai';
 import {Scanner} from '../../src/scanner';
-import {subscriptionId} from '.';
+import {subscriptionId, resourceGroup} from '.';
+import {RuleContext} from '../../src/rules';
 
 describe('Scanner', function () {
   this.slow(5000);
   this.timeout(8000);
+  const scanner = new Scanner();
   it('can load and execute resource graph rules from a JSON file', async () => {
-    const scanner = new Scanner();
-    await scanner.loadRulesFromFile('../test/rules.json');
-    const rgResults = await scanner.scan('resourceGraph', subscriptionId);
-    assert.equal(rgResults.length, 1);
-    rgResults.forEach(r => {
-      assert.containsAllKeys(r, ['ruleName', 'description', 'total', 'ids']);
-      assert.equal(r.total, 0);
+    const ruleContext = await scanner.getRulesFromFile(
+      'resourceGraph',
+      undefined,
+      '../test/rules.json'
+    );
+    const target = {subscriptionId};
+    const results = await scanner.scan(ruleContext, target);
+    assert.equal(results.length, ruleContext.rules.length);
+    results.forEach(r => {
+      assert.containsAllKeys(r, [
+        'ruleName',
+        'description',
+        'total',
+        'resources',
+      ]);
     });
-    const dummyResults = await scanner.scan('dummy', '');
-    assert.equal(dummyResults.length, 2);
-    dummyResults.forEach(r => {
-      assert.containsAllKeys(r, ['ruleName', 'description', 'total', 'ids']);
-      assert.equal(r.total, 0);
+  });
+
+  it('can scan a execute a resource graph rule against a subscription', async () => {
+    const rule = {
+      name: 'get-vnets',
+      description: 'gets all vnets in a resource group',
+      query: `Resources | where type =~ 'Microsoft.Network/virtualNetworks' | where resourceGroup=='${resourceGroup}' and name == 'vnet'`,
+    };
+    const ruleContext: RuleContext = {
+      type: 'resourceGraph',
+      target: {subscriptionId},
+      rules: [rule],
+    };
+
+    const results = await scanner.scan(ruleContext, ruleContext.target);
+    assert.equal(results.length, 1);
+    results.forEach(r => {
+      assert.equal(r.ruleName, rule.name);
+      assert.equal(r.description, rule.description);
+      assert.equal(r.total, 1);
+      assert.equal(r.resources.length, 1);
+    });
+  });
+  it('can execute a resource graph rule against a subscription and resource group', async () => {
+    const ruleContext = await scanner.getRulesFromFile(
+      'resourceGraph',
+      ['get-vnets'],
```

Comment:
This method doesn't feel nice to use for the happy path: just running all the rules in a file. I have to pass in something here that I don't really care about - a null/undefined/empty array, etc.